src/Simulation/Core/Attributes.cs

```diff
@@ -6,6 +6,23 @@
 
 namespace Microsoft.Quantum.Simulation.Core
 {
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    public class SpecializationRangeAttribute : Attribute
```

Comment:
nit: would `SpecializationSourceLocationAttribute` or just `SourceLocation` be a better name?

Reply:
Yes, renamed it to SourceLocationAttribute.